Source/CesiumRuntime/Public/CesiumGeoreference.h

```diff
@@ -129,473 +122,597 @@ class CESIUMRUNTIME_API ACesiumGeoreference : public AActor {
+   * Earth-Centered, Earth-Fixed (ECEF) coordinates.
    */
-  static FName DEFAULT_GEOREFERENCE_TAG;
+  UFUNCTION(
+      BlueprintPure,
+      Category = "Cesium",
+      meta =
+          (DeprecatedFunction,
+           DeprecationMessage =
+               "Use LongitudeLatitudeHeightToEarthCenteredEarthFixed on CesiumWgs84Ellipsoid instead."))
+  FVector TransformLongitudeLatitudeHeightToEcef(
+      const FVector& LongitudeLatitudeHeight) const;
 
   /**
-   * The percentage scale of the globe in the Unreal world. If this value is 50,
-   * for example, one meter on the globe occupies half a meter in the Unreal
-   * world.
+   * Transforms the given Earth-Centered, Earth-Fixed (ECEF) coordinates into
+   * WGS84 longitude in degrees (x), latitude in degrees (y), and height above
+   * the ellipsoid in meters (z).
    */
-  UPROPERTY(
-      EditAnywhere,
-      Meta = (AllowPrivateAccess),
-      BlueprintReadWrite,
-      BlueprintSetter = "SetScale",
-      BlueprintGetter = "GetScale",
+  UFUNCTION(
+      BlueprintPure,
       Category = "Cesium",
-      Meta = (UIMin = 0.000001, UIMax = 100.0))
-  double Scale = 100.0;
+      meta =
+          (DeprecatedFunction,
+           DeprecationMessage =
+               "Use EarthCenteredEarthFixedToLongitudeLatitudeHeight on CesiumWgs84Ellipsoid instead."))
+  FVector TransformEcefToLongitudeLatitudeHeight(const FVector& Ecef) const;
 
   /**
-   * The radii, in x-, y-, and z-direction, of the ellipsoid that
-   * should be used in this instance.
+   * Computes the rotation matrix from the local East-North-Up to
+   * Earth-Centered, Earth-Fixed (ECEF) at the specified ECEF location.
    */
-  UPROPERTY()
-  double _ellipsoidRadii[3];
-
-  GeoTransforms _geoTransforms;
+  UFUNCTION(
+      BlueprintPure,
+      Category = "Cesium",
+      meta =
+          (DeprecatedFunction,
+           DeprecationMessage =
+               "Use EastNorthUpToEarthCenteredEarthFixed on CesiumWgs84Ellipsoid instead."))
+  FMatrix ComputeEastNorthUpToEcef(const FVector& Ecef) const;
 
-  UPROPERTY()
-  UCesiumSubLevelSwitcherComponent* SubLevelSwitcher;
+#pragma endregion
 
-#if WITH_EDITOR
-  FDelegateHandle _newCurrentLevelSubscription;
-#endif
+#pragma region Implementation Details
 
-  // TODO: add option to set georeference directly from ECEF
-  void _setGeoreferenceOrigin(
-      double targetLongitude,
-      double targetLatitude,
-      double targetHeight);
+public:
+  ACesiumGeoreference();
 
-#if WITH_EDITOR
   /**
-   * Will make sure that the `CesiumSubLevels` array contains all
-   * of the current streaming levels of the world.
+   * Recomputes all world georeference transforms. Usually there is no need to
+   * explicitly call this from external code.
    */
-  void _updateCesiumSubLevels();
-#endif
-
-#if WITH_EDITOR
-  void _lineTraceViewportMouse(
-      const bool ShowTrace,
-      bool& Success,
-      FHitResult& HitResult) const;
+  void UpdateGeoreference();
```

Comment:
Can we make this private scope? The comment states that there is usually no need to call from external code and I don't see any occurrences that it is actually called from outside of this class.

Reply:
Awesome. I hope no one felt the need to call this too.